test/integration/controller_instance_test.go

```diff
@@ -1540,3 +1540,33 @@ func TestPollServiceInstanceLastOperation(t *testing.T) {
 		})
 	}
 }
+
+// TestRetryAsyncDeprovision tests whether asynchronous deprovisioning retries
+// by attempting a new deprovision after failing.
+func TestRetryAsyncDeprovision(t *testing.T) {
```

Comment:
I feel like I'm missing something when trying to understand this test. Where exactly is the deprovision reattempted, and where are the asserts? This is for my own learning, not really a review. 😊

Reply:
`ct.run` will by default implictly go though the standard process of creating a broker, provisioning, binding, unbinding, and deprovisioning, making sure that each finishes successfully. You can tweak the broker's reactions to this by modifying our fake osb client `ct.osbClient`'s `XReaction` objects.

So what I basically did was, in `setup`, program the broker's reactions to respond to the deprovision request with an async operation, then fail it in the poll, and then succeed the next deprovision request.

Reply:
Aha! I see it now! 😀 Thank you for the explanation.